policies/lib/tree/30_generic_methods/file_from_template_options.cf

```diff
@@ -0,0 +1,105 @@
+# @parameter            template_string String containing the template to be rendered
+# @parameter_constraint template_string "allow_empty_string" : true
+# @parameter            template_path  Source file containing a template to be rendered (absolute path on the target node)
+# @parameter_constraint template_path "allow_empty_string" : true
+# @parameter            show_content Show the file content in the report (defaults to true)
+# @parameter_constraint show_content "allow_empty_string" : true
+# @parameter_constraint show_content "select" : [ "", "true", "false" ]
+#
+# @class_prefix file_from_template_options
+# @class_parameter destination
+# @agent_support = ["cfengine-community"]
+
+promise agent template {
+  path => "/opt/rudder/bin/rudder-module-template";
+}
+
+bundle agent file_from_template_options(destination, engine, data, template_string, template_path, show_content) {
+  vars:
+      "module_file"  string => "/opt/rudder/bin/rudder-module-template";
+      "class_prefix" string => canonify("file_from_template_options_${destination}");
+
+  defaults:
+      "engine" string => "minijinja", if_match_regex => "";
+      "show_content" string => "true", if_match_regex => "";
+
+  classes:
+      "has_module"   expression => fileexists("${module_file}");
+
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+  methods:
+    pass3.has_module::
+      "${report_data.method_id}" usebundle => call_method("rudder_actual_file_from_template_options");
+      "${report_data.method_id}" usebundle => rudder_actual_file_from_template_options("${destination}", "${engine}", "${data}", "${template_string}", "${template_path}", "${show_content}");
+      "${report_data.method_id}" usebundle => call_method_classes("${class_prefix}");
+      "${report_data.method_id}" usebundle => call_method_classes_caller;
+      "${report_data.method_id}" usebundle => call_method_end("rudder_actual_file_from_template_options");
+      "${report_data.method_id}" usebundle => log_rudder_v4("${destination}", "Template rendering of ${destination}", "");
+
+    pass3.!has_module::
+      "${report_data.method_id}" usebundle => _classes_failure("${class_prefix}");
+      "${report_data.method_id}" usebundle => _classes_failure("${report_data.method_id}");
+      "${report_data.method_id}" usebundle => log_rudder_v4("${destination}", "Template module is not supported on this agent", "");
+}
+
+bundle agent rudder_actual_file_from_template_options(destination, engine, data, template_string, template_path, show_content) {
+  vars:
+   # "datastate" string => string_mustache("{{%-top-}}"),
+   #			unless => isvariable($(this.promiser));
+   "datastate_path" string => "/var/rudder/tmp/datastate-${this.promiser_pid}.json";
+
+  classes:
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+  files:
```

Comment:
The removal of the datastate file is missing. See https://github.com/Normation/rudder/blob/11ef4db06f575f104b91690682abb30a5a607f3d/policies/lib/tree/30_generic_methods/file_from_template_jinja2.cf#L291